### DISCUSSION TITLE
Podman behavior with hostip empty vs 0.0.0.0

### DISCUSSION THREAD

**Discussion by gb-nb** [containers/podman#20331] [Q&A]
Hello, I've been trying to use Testcontainers (helpful library to setup simulated environments from code) with podman. But I could not get it to work. It seems to be related to the testcontainers package always setting the `HostIp` to `"0.0.0.0"` instead of `""`(empty), when creating a container.

Also see this discussion for more information: https://github.com/testcontainers/testcontainers-dotnet/discussions/1014

Now I've read that empty means dual stack, so binding to both ipv6 and ipv4 stack. Where `0.0.0.0` should only bind the ipv4 stack. See here: https://github.com/containers/podman/issues/17780#issuecomment-1658194953

The thing is with the empty `HostIp` configuration I can reach my container just fine, but with `0.0.0.0` binding it is unreachable. Should I consider this a bug with podman, or might it be WSL / Windows related, or anything specific with routing behavior.

Currently using:
- Windows 10 22H2 19045.3448
- Podman 4.6.2

**Comment by Luap99** [comment]
There is some special magic on WSL port handling and overall WSL network stack seems to cause some troubles.

How do you run podman on WSL, are you using podman machine? Does podman run as root or rootless?

**Reply by Luap99** [reply]
we have some workaround but that only applies to rootless podman started via podman machine AFAICT: https://github.com/containers/podman/blob/main/cmd/rootlessport/wsl.go

**Reply by gb-nb** [reply]
Thank you, for your quick response. It is a while ago I first installed podman, but I checked my machine config, and it has `"Rootful": true,`.

I'm using `podman machine`. So given your second reply, I might try configuring a new podman machine which is rootless?

**Reply by Luap99** [reply]
Yes that might make a difference although I am not sure if/how this works on WSL.
I think @n1hility should know more about the differences

**Reply by gb-nb** [reply]
Switching to rootless fixed it! Thanks!

Just did the following steps:
```
> podman machine stop
Machine "podman-machine-default" stopped successfully
> podman machine set --rootful=false
> podman machine start
Starting machine "podman-machine-default"
```
